sw/device/lib/testing/i2c_testutils.c

```diff
@@ -0,0 +1,77 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/i2c_testutils.h"
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_i2c.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+
+#include "i2c_regs.h"  // Generated.
+
+static const uint8_t kI2cWrite = 0;
+static const uint8_t kI2cRead = 1;
+
+// Default flags for i2c operations.
+static const dif_i2c_fmt_flags_t kDefaultFlags = {.start = false,
+                                                  .stop = false,
+                                                  .read = false,
+                                                  .read_cont = false,
+                                                  .suppress_nak_irq = false};
+
+void i2c_testutils_wr(dif_i2c_t *i2c, uint8_t addr, uint8_t byte_count,
+                      uint8_t *data, bool skip_stop) {
+  dif_i2c_fmt_flags_t flags = kDefaultFlags;
+  uint8_t data_frame;
+
+  // TODO, the current function does not support write payloads
+  // larger than the fifo depth.
+  CHECK(byte_count < I2C_PARAM_FIFO_DEPTH);
+
+  // First write the address.
+  flags.start = true;
+  LOG_INFO("testutils: address %d address %d", addr, addr < 1);
+  data_frame = (addr << 1) | kI2cWrite;
+  CHECK_DIF_OK(dif_i2c_write_byte_raw(i2c, data_frame, flags));
+
+  // Once address phase is through, blast the rest as generic data.
+  flags = kDefaultFlags;
```

Comment:
The variable `flags` hasn't changed since the initialization, so I would drop this line.

Reply:
wait it has right? in line 39 the "start" field was set to true.